beacon-chain/blockchain/process_block.go

```diff
@@ -423,11 +381,29 @@ func (s *Service) onBlockBatch(ctx context.Context, blks []interfaces.SignedBeac
 			JustifiedCheckpoint: jCheckpoints[i],
 			FinalizedCheckpoint: fCheckpoints[i]}
 		pendingNodes[len(blks)-i-1] = args
-		s.saveInitSyncBlock(blockRoots[i], b)
-		if err = s.handleBlockAfterBatchVerify(ctx, b, blockRoots[i], fCheckpoints[i], jCheckpoints[i]); err != nil {
+		if err := s.saveInitSyncBlock(ctx, blockRoots[i], b); err != nil {
 			tracing.AnnotateError(span, err)
 			return err
 		}
+		if err := s.cfg.BeaconDB.SaveStateSummary(ctx, &ethpb.StateSummary{
+			Slot: b.Block().Slot(),
+			Root: blockRoots[i][:],
+		}); err != nil {
+			tracing.AnnotateError(span, err)
+			return err
+		}
+		if i > 0 && jCheckpoints[i].Epoch > jCheckpoints[i-1].Epoch {
```

Comment:
in the event we only have 1 block in the whole batch, would this method not be able to updated the justified checkpoint ?

Reply:
if we only have one block in a batch then the block is only inserted below in `InsertNode` which does the full justification update. This is only used for intermediate blocks because we don't pass the full beaconState to forkchoice.